### ISSUE TITLE
BUG: Policy on unsupported BLAS implementations

### ISSUE THREAD

**Issue by jbrockmendel** [statsmodels/statsmodels#5104]
TL;DR: if there are config/function pairs in which statsmodels is currently returning incorrect answers, those should either a) be made to return correct answers or b) explicitly raise rather than silently being incorrect

#4737 is the most recent of many places where differences between MKL vs openblas have had user-facing consequences (at least that is the leading hypothesis in that discussion).

The other place I've seen this is in the test suite where some `assert_allclose` checks seem less precise than they could be, the reasoning being that numerical noise makes more precise checks infeasible.

I'll take a look and see if there are other relevant Issues.  This will include general flaky-test cases that _may_ be related.
- [ ] #4737 
- [ ] #4779 
- [ ] #4643 Failures in TestHoltWinters.test_forecast
- [ ] #4396 test_compare_numdiff flaky test

**Comment by bashtage** [comment]
#4643 is a convergence failure that effectively adds some randomness that depends on the BLAS.  I wouldn't call it a BLAS issue.

**Comment by bashtage** [comment]
Seems to have improved recently, which makes sense since OpenBlas has been getting better.
